File: Domain/BeSpoked.BLL/Products/Entities/Product.cs
using BeSpoked.Common.EntityService.Entities;
using BeSpoked.Sales.Entities;

namespace BeSpoked.Products.Entities;

public record Product : Entity
{
    public required string Name { get; set; }
    public required string Manufacturer { get; set; }
    public ProductStyle Style { get; set; }
    public decimal PurchasePrice { get; set; }
    public decimal SalePrice { get; set; }
    public int QuantityOnHand { get; set; }
    public decimal CommissionPercentage { get; set; }
    public virtual ICollection<Sale> Sales { get; set; } = new HashSet<Sale>();
}

File: WebApp/ClientApp/src/Entities/Product.ts
export interface Product {
    id: string;
    name: string;
    manufacturer: string;
    style: ProductStyle;
    purchasePrice: number;
    salePrice: number;
    quantityOnHand: number;
    commissionPercentage: number;
}

export enum ProductStyle {
    Mountain = 1,
    Race = 2,
    Kids = 3,
}

export type ServerProduct = Product;

export function deserializeProduct(serverResponse:ServerProduct):Product{
    return {...serverResponse }
}